flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1176,8 +1177,22 @@ public void reportPayload(ResourceID resourceID, Void payload) {
 	//  Resource Management
 	// ------------------------------------------------------------------------
 
-	protected int getNumberPendingSlotRequests() {
-		return slotManager.getNumberPendingSlotRequests();
+	protected int getNumberRequiredTaskManagerSlots() {
+		return slotManager.getNumberPendingTaskManagerSlots();
+	}
+
+	// ------------------------------------------------------------------------
+	//  Helper methods
+	// ------------------------------------------------------------------------
+
+	protected static Collection<ResourceProfile> createSlotsPerWorker(int numSlots) {
+		final Collection<ResourceProfile> slots = new ArrayList<>(numSlots);
```

Comment:
It would be enough to write:
`Collections.nCopies(numSlots, ResourceProfile.ANY);` 

The returned `List` is also immutable. At the moment there is no need to return a mutable list. I would even say it is not right that after calling `Collection<ResourceProfile> startNewWorker(ResourceProfile resourceProfile)`, one may potentially modify the state of the ResourceManager.

Reply:
Good point. I'll change it accordingly.

Reply:
Good point. I'll update the PR.